src/utils/image_normalizer.py

```diff
@@ -36,16 +36,25 @@ def build_output_path(self, path):
     # change the extension to jpg
     return dest.with_suffix('.jpg')
 
-  # Resizes the provided image based on the longest dimension, if it exceeds the provided max_dimension.
+  # Resizes the provided image based on:
+  #   1. the longest dimension, if it exceeds the provided max_dimension.
+  #   2. the shortest dimension, if it would be less than min_dimension after resizing based on longest dimension
   # Otherwise, returns the original image.
-  def resize_to_max_dimension(img, max_dimension):
+  def resize(self, img):
     width, height = img.width, img.height
-    if max(width, height) <= max_dimension:
+    max_dimension, min_dimension = self.config.max_dimension, self.config.min_dimension
+    if max(width, height) <= max_dimension or min(width, height) <= min_dimension:
```

Comment:
Is this returning the image if it's smaller than the minimum dimensions? There seem to be checks that can't be reached to resize the image if that's the case.

Reply:
yes, if the shortest dimension is less than or equal to the minimum dimension property it returns the original. I think all of the branches of the conditionals further down are being reached in the tests. Are you concerned about the nested conditions that reset the size if the shortest edge post-resizing becomes less than the minimum, like here?
https://github.com/UNC-Libraries/ml-repo-preingest-processing/pull/5/files#diff-0c2a60b30159a6fa12e435f430820886a7384270eebc8344c33bc72d846011afR51-R53